botocore/docs/bcdoc/docstringparser.py

```diff
@@ -129,29 +135,32 @@ def __init__(self, tag, attrs=None, parent=None):
         self.attrs = attrs
         self.tag = tag
 
-    def write(self, doc):
+    def write(self, doc, next):
         self._write_start(doc)
         self._write_children(doc)
-        self._write_end(doc)
+        self._write_end(doc, next)
 
     def _write_start(self, doc):
         handler_name = 'start_%s' % self.tag
         if hasattr(doc.style, handler_name):
             getattr(doc.style, handler_name)(self.attrs)
 
-    def _write_end(self, doc):
+    def _write_end(self, doc, next):
         handler_name = 'end_%s' % self.tag
         if hasattr(doc.style, handler_name):
-            getattr(doc.style, handler_name)()
+            if handler_name == 'end_a':
+                getattr(doc.style, handler_name)(next)
```

Comment:
Fact check me, because I may be wrong here. We should only be passing in a non-None value here when we've:

1.) Determined we're in a Tag node
2.) We have a look ahead character

Would it be fair to say we could pass next to any handler if the value of next is not None?

Reply:
I dont think this would work since it would allow a `next_child` to be passed for other tag node's such as `end_i`, `end_ul`, `end_li`... which currently don'g accent a `next_child` since its not needed. Only `end_a` accepts this parameter.